### PR TITLE
Disable attempts increase when resetting devicelab task

### PR DESCRIPTION
**Pull request by keyonghan**
This PR fixes issue: https://github.com/flutter/flutter/issues/44780

### CODE REVIEW

File: app_dart/lib/src/request_handlers/reset_devicelab_task.dart [hunk 1]
```diff
@@ -35,8 +35,9 @@ class ResetDevicelabTask extends ApiRequestHandler<Body> {
   Future<Body> post() async {
     checkRequiredParameters(<String>[keyParam]);
     final String encodedKey = requestData[keyParam];
+    final ClientContext clientContext = authContext.clientContext;
     final KeyHelper keyHelper =
-        KeyHelper(applicationContext: context.applicationContext);
+        KeyHelper(applicationContext: clientContext.applicationContext);
```
Comment (CaseyHillers) [inline app_dart/lib/src/request_handlers/reset_devicelab_task.dart]: Why was this changed?

File: app_dart/test/request_handlers/reset_devicelab_task_test.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          key: commit.key.append(Task, id: 4590522719010816),
+          commitKey: commit.key,
+          attempts: 0);
+      config.db.values[task.key] = task;
+
+      expect(task.attempts, 0);
+
+      await tester.post(handler);
+      // Reset devicelab task will not increase [attempts].
+      expect(task.attempts, 0);
```
Comment (CaseyHillers) [inline app_dart/test/request_handlers/reset_devicelab_task_test.dart]: Can we add an expectation that the task is marked new?

Comment (keyonghan) [reply app_dart/test/request_handlers/reset_devicelab_task_test.dart]: Added.

File: app_dart/lib/src/request_handlers/reset_devicelab_task.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/lib/src/request_handlers/reset_devicelab_task.dart]: `context` tries to find `_APPENGINE_CONTEXT`, but test fails in local development.

Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/reset_devicelab_task.dart]: That's interesting. Is this how the other Authenticated RequestHandlers use keyhelper then?

Comment (keyonghan) [reply app_dart/lib/src/request_handlers/reset_devicelab_task.dart]: I believe so. If not, then we need to update when creating corresponding tests.
